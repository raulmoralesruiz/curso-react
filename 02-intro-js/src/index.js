console.log("sustituir el contenido de este fichero index.js por cualquier otro de las bases");
