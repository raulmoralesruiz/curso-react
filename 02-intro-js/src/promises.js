const promesa = new Promise((resolve, reject) => {
  setTimeout(() => {
    resolve("Hola Mundo");
  }, 2000);
});

promesa.then((mensaje) =>
  console.log(mensaje).catch((err) => console.warn(err))
);
