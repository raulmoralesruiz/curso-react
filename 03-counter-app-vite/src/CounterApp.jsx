import { useState } from "react";
import PropTypes from "prop-types";

// const click = (event) => {
//   console.log(event);
// };

export const CounterApp = ({ value }) => {

  const [counter, setCounter] = useState(value);
  
  const clickSum = () => {
    setCounter(counter + 1);
  };
  const clickSub = () => setCounter(counter - 1);
  const clickReset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>

      <button onClick={clickSum}>+1</button>
      <button onClick={clickSub}>-1</button>
      <button aria-label="btn-reset" onClick={clickReset}> Reset </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
