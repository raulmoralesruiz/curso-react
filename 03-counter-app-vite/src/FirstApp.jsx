import PropTypes from "prop-types";

export const FirstApp = ({ title, subtitle, name }) => {
  // console.log(props);

  return (
    <>
      <h1>{title}</h1>
      {/* <code>{ JSON.stringify(variable) }</code> */}
      <p>{subtitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  // title: "Default title",
  subtitle: "Default subtitle",
  name: "Default name",
};