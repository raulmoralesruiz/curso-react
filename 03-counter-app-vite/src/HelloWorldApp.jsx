// export const App = () => {
//   return <h1>Hello World</h1>;
// }

// export default App;


export const HelloWorldApp = () => {
  return (
    <h1>HelloWorld App</h1>
  )
}
