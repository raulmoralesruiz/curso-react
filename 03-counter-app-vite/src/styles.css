html, body {
  background-color: #21232A;
  color: white;
  font-family: Arial, Helvetica, sans-serif;
  font-size: 1.3rem;
  padding: 70px;
}

button {
  background-color: #43464d;
  border: none;
  color: white;
  cursor: pointer;
  font-size: 1.3rem;
  padding: 10px 20px;
  margin: 1px;
}