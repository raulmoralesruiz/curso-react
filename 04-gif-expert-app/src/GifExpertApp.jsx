import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([
    "One Punch",
    "Samurai X",
    "Dragon Ball",
  ]);

  const onAddCategory = ( newCategory ) => {
    // evitar repetir categorias
    if (categories.includes(newCategory)) return;

    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
