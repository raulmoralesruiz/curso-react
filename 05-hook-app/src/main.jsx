import React from "react";
import ReactDOM from "react-dom/client";
import { Memorize } from "./06-memos/Memorize";
// import { FocusScreen } from "./04-useRef/FocusScreen";
// import { Layout } from "./05-useLayoutEffect/Layout";
// import { CounterApp } from "./01-useState/CounterApp";
// import { CounterWithCustomHook } from "./01-useState/CounterWithCustomHook";
// import { FormWithCustomHook } from "./02-useEffect/FormWithCustomHook";
// import { MultipleCustomHooks } from "./03-examples/MultipleCustomHooks";
// import { SimpleForm } from "./02-useEffect/SimpleForm";
// import { HooksApp } from "./HooksApp";
import "./index.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  // <React.StrictMode>
    <Memorize />
  // </React.StrictMode>
);
