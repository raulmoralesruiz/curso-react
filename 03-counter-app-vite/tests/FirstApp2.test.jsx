import {render, screen} from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en <FirstApp />', () => { 

  const title = "Hola, soy Goku";
  const subtitle = "Soy un subtítulo";

  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar el mensaje 'Hola, soy Goku'", () => {
    render(<FirstApp title={title} />);
    expect( screen.getByText(title)).toBeTruthy();
    //screen.debug();
  });

  test("Debe mostrar el título en un h1", () => {
    render(<FirstApp title={title} />);
    expect( screen.getByRole('heading', {level: 1})).toBeTruthy();
    expect( screen.getByRole('heading', {level: 1}).innerHTML).toContain(title);
  });

  test("Debe mostrar el subtítulo enviado por props", () => {
    render(<FirstApp title={title} subtitle={subtitle} />);
    expect( screen.getByText(subtitle)).toBeTruthy();
  });
  
});