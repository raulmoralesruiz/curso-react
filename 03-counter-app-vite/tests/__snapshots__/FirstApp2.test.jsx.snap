// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola, soy Goku
  </h1>
  <p>
    Default subtitle
  </p>
  <p>
    Default name
  </p>
</div>
`;
